### PR TITLE
Add focus handling

### LINKED ISSUE #838: Add focus tracking
We are migrating from `react-redux-form` to `formik`. Several of our forms make use of the focus property in RRF for displaying errors and help information. 

It would be nice if Formik tracked which field was in focus as part of it's state. Something similar to the way blur/touched is tracked would work. Formik could provide `handleFocus` and `setFieldFocused` for setting the focus and a `focused` object that has the current field that is in focus set to `true` and all other known fields set to `false`.

Alternatively, we can implement a wrapper component to track focus outside of formik, but I think it would be cleaner if this functionality was implemented in formik.

### PR DESCRIPTION
**Pull request by AntiFish03**
Added focus handling using the touched/blur handling as a pattern.
This should resolve #838.

### CODE REVIEW

File: src/Formik.tsx [hunk 1]
```diff
@@ -503,6 +508,45 @@ export class Formik<Values = {}, ExtraProps = {}> extends React.Component<
...
+      if (!field && process.env.NODE_ENV !== 'production') {
+        warnAboutMissingIdentifier({
+          htmlContent: outerHTML,
+          documentationAnchorLink: 'handlefocus-e-any--void',
+          handlerName: 'handleFocus',
+        });
+      }
+
+      this.setState(prevState => ({
+        focused: setIn(prevState.focused, field, true),
```
Comment (ematipico) [inline src/Formik.tsx]: Only one field at the time can be focused. So all the rest of the fields should be turned to false

**Comment by AntiFish03** [comment]
@jaredpalmer Any comments/reviews on this PR?

**Comment by weyert** [comment]
Looks interesting PR to have in Formik!

**Comment by AntiFish03** [comment]
@rohit-sourcefuse I really don't want to try and maintain a fork of formik, however you can use my forks master branch, I have added a `prepare` lifecycle script to it to allow this to be built on `yarn install`/`npm install` to make it possible to use without a published package.

**Comment by AntiFish03** [comment]
@jaredpalmer Where were you back in Oct/Nov?  Forgive me for being blunt but there has been plenty of time for this to be merged (when it was in sync with your master) without additional work from me or anyone else.

**Comment by MontoyaAndres** [comment]
Is this function added? or when will be? I want to use it :smile:

**Comment by jaredpalmer** [comment]
This PR is actually a little overdone. Only one field can be focused at at a time.  We just need to keep the name of the currently focused field in Formik state and the set it to undefined within handleBlur. A more advanced solution is in the works with v2 whereby we register refs so you can manually switch focus. This is very very useful for React Native.

**Comment by MastaDon63** [comment]
I Would love to get started using this or a solution like this. I see that v2 is in alpha, is there any kind of firm timeline for v2? If not would it be possible to use this implementation in the meantime?

**Comment by kaifaust** [comment]
Trying to implement floating labels, would have been great if this was done by now.

**Comment by jaypatel512** [comment]
Can this PR somehow be revived. I think a better customer experience bringing focus back on the error input is a must these days. And this PR has a potential if some love is given to it. I hope we can use this PR for v1, and use the newer simpler approach in V2 when it matures.

**Comment by williampruden** [comment]
@jaredpalmer so did this feature finally make it in or?

**Comment by jaredpalmer** [comment]
I’m going to be adding it on my own in a different PR.

**Comment by Morkowski** [comment]
👍

**Comment by alanrubin** [comment]
Any news on that? I'm looking forward to it so that I can avoid showing errors if a field is focused and being worked in.

**Comment by l0gicgate** [comment]
Why was this PR closed instead of just fixing the outstanding issues? There are no other PRs that supersede this.

**Comment by johnrom** [comment]
PR that supersedes this: https://github.com/jaredpalmer/formik/pull/2317

**Comment by l0gicgate** [comment]
Thank you @johnrom!

**Pull request closed without merging**